core/store/package_manifest.rs

```diff
@@ -83,6 +84,26 @@ impl PackageManifest {
         self.published_at
     }
 
+    pub async fn write(&self, path: &Path) -> Result<(), FromFileError> {
+        let json = serde_json::to_string_pretty(&self).unwrap();
+        fs::write(&path, json)
+            .await
+            .map_err(|err| FromFileError::CouldNotWriteFile {
+                err,
+                file: path.to_path_buf(),
+            })
+    }
+
+    // replaces the existing keys with those of a new manifest
+    pub fn merge(self, other: Self) -> Self {
+        let mut keys = self.keys().clone();
```

Comment:
We don't need to clone here. `self` can be declared as `mut self` and then we can call `self.keys.insert` directly.

Similarly for `other.keys().iter()`, you can call `other.keys.into_iter()` . Notice 2 things:
1. how we skip the accessor function, since we're within the struct we have access to it
2. the call `into_iter()` takes a collection and turns it into an _owned iterator_, instead of a _borrowed iterator_.

This will remove the rest of the clones.

Then at the bottom you can just return `self` since that will have been updated.